Common/Statistics/AssetsUnderManagementCapacityManager.cs

```
@@ -0,0 +1,238 @@
+
+                    if (holdingsValue != 0)
+                    {
+                        var order = _portfolio.Transactions.GetOrderById(orderId);
+                        var orderValue = Math.Abs(order.GetValue(security));
+                        holdingsTurnover = orderValue / holdingsValue;
+
+                        if (holdingsTurnover == 0)
+                        {
+                            RemoveConsolidator(sender, symbol);
+                            return;
+                        }
+                    }
+
+                    var totalMarketVolume = GetTotalMarketVolume(data);
+                    var totalTradeVolumeCapacity = _maximumTradeableVolume * totalMarketVolume / holdingsTurnover;
+
+                    // If it is a new day, we discard the previous day data
+                    var utcDate = data.EndTime.ConvertToUtc(security.Exchange.TimeZone).Date;
+                    if (_previousInputTime != utcDate)
+                    {
+                        _previousInputTime = utcDate;
+                        _maximumSymbolCapacity.Clear();
+                        _historicalPortfolioCapacity.Add(0m);
+                    }
+
+                    // Updates AUM Capacity if there is new information or lower total trade volume capacity for the security
+                    decimal capacity;
+                    if (!_maximumSymbolCapacity.TryGetValue(symbol, out capacity) ||
+                        totalTradeVolumeCapacity < capacity)
+                    {
+                        _maximumSymbolCapacity[symbol] = totalTradeVolumeCapacity;
+                        _historicalPortfolioCapacity[0] = _maximumSymbolCapacity.Sum(x => x.Value);
+                    }
+
+                    RemoveConsolidator(sender, symbol);
+                }
+                catch (Exception e)
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager.Update: {e}");
```

There seems to be a chance of an exception being thrown and the consolidator not being removed, maybe we can move the `RemoveConsolidator(sender, symbol);` for it to be the first line in the `try/catch` block?

---

Also, please pass the error object as it's own argument to `Log.Error` to ensure the full exception object makes it down to the configured logging framework. For QC, we end up doing something similar to what you have here, but I know the community has made adapters for various logging framework, such as log4net, which greatly benefit from having access to the raw exception object.

Another also, the catch all exception handling makes me uneasy. Is this necessary? Why not just let it throw an exception? Is there potential for a non-fatal error being raised?